### PR TITLE
added codim one hyper ball

### PR DESCRIPTION
**Pull request by nicolacavallini**
I added codimension one for 
hyper_ball<1,2> and hyper_ball<2,3>

### CODE REVIEW

File: include/deal.II/grid/grid_generator.h [hunk 1]
```diff
@@ -358,6 +358,19 @@ namespace GridGenerator
...
+    * This function is declared to exist for dim=1, spacedim =2, and
+    * dim=2, spacedim =3.
+    %
+    * @note The triangulation needs to be void upon calling this function.
+    */
+
+  template <int dim, int spacedim>
+  void hyper_ball (Triangulation<dim,spacedim> &tria,
+                   const Point<spacedim>   &center = Point<spacedim>(),
+                   const double        radius = 1.);
```
Comment (bangerth) [inline include/deal.II/grid/grid_generator.h]: Why don't you make this the only declaration available? In other words, give the original declaration a second template argument `spacedim`?

File: source/grid/grid_generator.cc [hunk 2]
```diff
@@ -2861,7 +2861,19 @@ namespace GridGenerator
...
+  hyper_ball (Triangulation<dim,spacedim> &tria,
+              const Point<spacedim>   &p,
+              const double      radius)
+  {
+    Triangulation<spacedim> volume_mesh;
+    GridGenerator::hyper_ball(volume_mesh,p,radius);
+    std::set<types::boundary_id> boundary_ids;
+    boundary_ids.insert (0);
+    GridGenerator::extract_boundary_mesh (volume_mesh, tria,
+                                          boundary_ids);
```
Comment (bangerth) [inline source/grid/grid_generator.cc]: uh oh, I now understand. When you say `hyperball<2,3>` you mean the *surface* of the ball, right? I think then you're using the wrong word. In mathematical English, a *ball* are all the points `||x|| <= r` whereas a *sphere* are all the points with `||x||=r`. In other words, what you're creating here is a hypersphere. I think that's how you should name the function.

File: tests/grid/grid_generator_hyper_ball_codim_one.cc [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+
+// Test GridGenerator::create_triangulation_with_removed_cells
```
Comment (bangerth) [inline tests/grid/grid_generator_hyper_ball_codim_one.cc]: need to adjust this comment

**Comment by bangerth** [comment]
Please adjust the name of the function. Do you happen to have pictures of the meshes? If so, we could also embed them into the documentation.

**Comment by nicolacavallini** [comment]
Hi!  Thanks for the comments!

Adjusted function name.

Is this the cosset location to place my pics?
dealii/doc/doxygen/images/

I include them with:

@image html pic_name.png

right?

**Comment by nicolacavallini** [comment]
guys somebody included bad indentation to:
(this is the git status after running the indentation script)

	modified:   source/base/function_lib.cc
	modified:   tests/base/functions_10.cc

travis is complaining

File: include/deal.II/grid/grid_generator.h [hunk 4]
```diff
@@ -358,6 +358,38 @@ namespace GridGenerator
                    const double        radius = 1.);
 
   /**
+    * Creates an hyper sphere.
+    * This function is declared to exist for dim=1, spacedim =2, and
+    * dim=2, spacedim =3.
+    *
+    * The following pictures are genreted with:
```
Comment (bangerth) [inline include/deal.II/grid/grid_generator.h]: genreted -> generated

File: include/deal.II/grid/grid_generator.h [hunk 5]
```diff
@@ -358,6 +358,38 @@ namespace GridGenerator
...
+    *
+    * GridGenerator::hyper_sphere(triangulation);
+    *
+    * triangulation.set_all_manifold_ids(0);
+    * triangulation.set_manifold (0, surface_description);
+    * triangulation.refine_global(3);
+    * @endcode
+    *
+    * See the @ref manifold "documentation module on manifolds", for
+    * bettere detail.
```
Comment (bangerth) [inline include/deal.II/grid/grid_generator.h]: bettere -> more

**Comment by bangerth** [comment]
Thanks for renaming and adding the pictures. Can you please fix the two typos?

**Comment by nicolacavallini** [comment]
Typos fixed.

**Comment by tamiko** [comment]
Intel is a bit picky about a minor issue [1]. :wink:

[1] http://cdash.kyomu.43-1.org/viewBuildError.php?type=1&onlydeltap&buildid=12589

File: source/grid/grid_generator.cc [hunk 6]
```diff
@@ -4317,6 +4331,18 @@ namespace GridGenerator
 }
 
 // explicit instantiations
+namespace GridGenerator
+{
+
+  template void
+  hyper_sphere< 1 ,  2 > (Triangulation< 1 ,   2> &,
+                          const Point<2>   &center = Point<2>(),
+                          const double        radius = 1.);
```
Comment (bangerth) [inline source/grid/grid_generator.cc]: Can you come up with a follow up patch that removes the default values of the arguments here and in the next couple of lines to silence the earnings @tamiko observes?

**Pull request merged**